tests/scalers/gitlab_runner/gitlab_runner_test.go

```diff
@@ -0,0 +1,503 @@
+`
+	scaledObjectTemplate = `
+apiVersion: keda.sh/v1alpha1
+kind: ScaledObject
+metadata:
+  name: {{.ScaledObjectName}}
+  namespace: {{.TestNamespace}}
+spec:
+  scaleTargetRef:
+    name: {{.DeploymentName}}
+  minReplicaCount: {{.MinReplicaCount}}
+  maxReplicaCount: {{.MaxReplicaCount}}
+  pollingInterval: 15
+  cooldownPeriod: 5
+  triggers:
+  - type: gitlab-runner
+    metadata:
+      gitlabAPIURL: {{.GitlabAPIURL}}
+      projectID: "{{.ProjectID}}"
+      targetPipelineQueueLength: "{{.TargetPipelineQueueLength}}"
+      labels: "e2eSOtester"
+    authenticationRef:
+     name: gitlab-trigger-auth
+`
+)
+
+func TestScaler(t *testing.T) {
+	// setup
+	t.Log("--- setting up ---")
+
+	t.Log("--- deleting all projects ---")
+	err := deleteAllUserProjects(gitlabBaseURL)
+	require.NoError(t, err)
+
+	t.Log("--- creating new project ---")
+	projectID, err := createNewProject(gitlabBaseURL)
+	require.NoError(t, err)
+
+	defer func() {
+		t.Log("--- cleanup project ---")
+		err := deleteRepo(gitlabBaseURL, projectID)
+		require.NoError(t, err)
+	}()
+
+	t.Log("--- add ci file ---")
+	err = commitFile(gitlabBaseURL, ciFileContent, projectID, ".gitlab-ci.yml", true)
+	require.NoError(t, err)
+
+	// Create kubernetes resources
+	t.Log("--- create kubernetes resources ---")
+	kc := GetKubernetesClient(t)
+	data, templates := getTemplateData(projectID)
+	CreateKubernetesResources(t, kc, testNamespace, data, templates)
+
+	WaitForPodCountInNamespace(t, kc, testNamespace, minReplicaCount, 60, 2)
+
+	// test scaling Scaled Object
+	KubectlApplyWithTemplate(t, data, "scaledObjectTemplate", scaledObjectTemplate)
+	testSONotActivated(t, kc)
+
+	testSOScaleOut(t, kc, projectID)
+	testSOScaleIn(t, kc, projectID)
+
+	// cleanup
+	DeleteKubernetesResources(t, testNamespace, data, templates)
+}
+
+func queueRun(t *testing.T, projectID string) {
+	err := commitFile(gitlabBaseURL, "dummy content hello world", projectID, "dummy"+uuid.NewString(), false)
+	require.NoError(t, err)
+}
+
+func getTemplateData(projectID string) (templateData, []Template) {
+	base64Pat := base64.StdEncoding.EncodeToString([]byte(personalAccessToken))
+
+	return templateData{
+			TestNamespace:             testNamespace,
+			SecretName:                secretName,
+			DeploymentName:            deploymentName,
+			ScaledObjectName:          scaledObjectName,
+			MinReplicaCount:           fmt.Sprintf("%v", minReplicaCount),
+			MaxReplicaCount:           fmt.Sprintf("%v", maxReplicaCount),
+			Pat:                       base64Pat,
+			GitlabAPIURL:              "https://gitlab.com",
+			ProjectID:                 projectID,
+			TargetPipelineQueueLength: "1",
+		}, []Template{
+			{Name: "secretTemplate", Config: secretTemplate},
+			{Name: "authTemplate", Config: triggerAuthTemplate},
+			{Name: "deploymentTemplate", Config: deploymentTemplate},
+			{Name: "scaledObjectTemplate", Config: scaledObjectTemplate},
+		}
+}
+
+func testSONotActivated(t *testing.T, kc *kubernetes.Clientset) {
```

Comment:
During the activation test, we have to add an item in the queue and check that the activation value blocks the scaling until the item count in the queue is higher